main_file.py

```diff
@@ -30,12 +30,18 @@
 import ICVL
 import MNIST
 
+from unpack_files import unpackFiles
 from shared_vae_class import shared_vae_class
 from model_objects import model_parameters
 # MNIST 28x28
 # ICVL 60x80
 
-dataSetInfo = ICVL.dataInfo()
+dataSetInfo = MNIST.dataInfo()
+
+if not os.path.exists(os.path.join('Data', 'Training',
+                                   '{}_Training.pkl'.format(dataSetInfo.
+                                                            name))):
+    unpackFiles(dataSetInfo.name)
```

Comment:
this will read and write files each time it is called?

Reply:
Only if the file does not already exist. i.e. it only writes the file if the files needs to be written.